src/pagination-with-raw.ts

```diff
@@ -0,0 +1,27 @@
+import { IPaginationLinks, IPaginationMeta } from './interfaces';
+import { Pagination } from './pagination';
+
+export class PaginationWithRaw<PaginationObject> extends Pagination<
+  PaginationObject
+> {
+  constructor(
+    /**
+     * a list of items to be returned
+     */
+    public readonly items: PaginationObject[],
+    /**
+     * a list of raw items when queried raw with entities
+     */
+    public readonly rawItems: any[],
```

Comment:
Just thinking, perhaps rawItems should not be provided on the pagination object? Trying to avoid someone returning rawItems in their response. Perhaps `paginateRawAndEntities` should return type `[Pagination, Partial<T>[]]` where `Partial<T>[]` is the raw results. 
I guess it's not really our responsibility to avoid this however we could mitigate the use of the second pagination object creation. What do you think?

Reply:
That way we could do something like this 

```ts
const [pagintation, raw] = paginateRawAndEntities<Entity, RawEntity>(queryBuilder); // returns [Pagination<Entity>, RawEntity[]]

pagination.items.map((item, index) => {
  item['someProperty'] = raw[index].someProperty;
});

return pagination;
```

Reply:
I was having a hard time figuring out how to return the values in a way that maintained pagination consistent, and I'm liking this solution

Reply:
cool :) I think other than this bit it's perfect!